test/sql/test_deprecations.py

```diff
@@ -135,6 +153,105 @@ def test_table_useexisting(self):
                 autoload_with=testing.db,
             )
 
+    def test_case_sensitive(self):
+        reg = functions._registry
+        cs_reg = functions._case_sensitive_reg
+
+        class MYFUNC(GenericFunction):
+            type = DateTime
+
+        assert isinstance(func.MYFUNC().type, DateTime)
+        assert isinstance(func.MyFunc().type, DateTime)
+        assert isinstance(func.mYfUnC().type, DateTime)
+        assert isinstance(func.myfunc().type, DateTime)
+
+        assert "myfunc" in reg['_default']
+        assert "MYFUNC" not in reg['_default']
+        assert "MyFunc" not in reg['_default']
+        assert "myfunc" in cs_reg['_default']
+        assert cs_reg['_default']['myfunc'] == ['MYFUNC']
+
+        with testing.expect_deprecated():
+            class MyFunc(GenericFunction):
+                type = Integer
+
+        assert isinstance(func.MYFUNC().type, DateTime)
+        assert isinstance(func.MyFunc().type, Integer)
+        with pytest.raises(AssertionError):
+            assert isinstance(func.mYfUnC().type, Integer)
+        with pytest.raises(AssertionError):
+            assert isinstance(func.myfunc().type, Integer)
+
+        assert "myfunc" not in reg['_default']
+        assert "MYFUNC" in reg['_default']
+        assert "MyFunc" in reg['_default']
+        assert "myfunc" in cs_reg['_default']
+        assert cs_reg['_default']['myfunc'] == ['MYFUNC', 'MyFunc']
```

Comment:
**mike bayer** (zzzeek) wrote:

for this one we use sqalchemy.testing.eq_(), that one is pretty consistent.

Reply:
Ok